osquery/tables/system/darwin/diagnostics.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <ctime>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/utils/conversions/split.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+// read data from core_analytics file
+void genDiagnosticsFromFile(const boost::filesystem::path& diagnostics_file,
+                            QueryData& results) {
+  std::string diagnostics_content;
+
+  if (forensicReadFile(diagnostics_file, diagnostics_content).ok()) {
```

Comment:
I'm not sure what our convention is, but looking at the codebase, it looks like we more commonly do this with an early return. For example https://github.com/osquery/osquery/blob/e6fe15eb49660725e65dba1549932ed96e0a8c6e/osquery/tables/system/posix/sudoers.cpp#L43-L47